src/components/pages/AddProduct/index.tsx

```diff
@@ -0,0 +1,311 @@
+import React, { FC, useMemo } from 'react';
+import { useAppDispatch, useAppSelector } from '../../../hooks/redux.ts';
+import { productSlice } from '../../../store/reducers/productsSlice.ts';
+import Select from 'react-select';
+import {
+  mainCategories,
+  subCategory,
+  unit,
+} from '../../../constants/categories.ts';
+import UploadAndDisplayImage from '../../common/FileUpload';
+import Checkbox from '../../common/Checkbox';
+import RegistrationField from '../../common/RegistrationField';
+import Button from '../../common/Button';
+
+const inputStyle =
+  'w-full border border-gray-200 rounded p-3 outline-none focus:bg-none focus:ring-0';
+
+const Label = ({ label }: { label: string }) => (
+  <p className={'text-[16px] font-medium mb-2 text-text_com leading-6'}>
+    {label}
+  </p>
+);
+
+export const AddProduct: FC = () => {
+  const { addProduct } = useAppSelector(state => state.productState);
+  const { user } = useAppSelector(state => state.userState);
+  const dispatch = useAppDispatch();
+
+  const { SET_FIELD } = productSlice.actions;
+
+  const subCategoryLocal = useMemo(() => {
+    return subCategory.filter(
+      el => el?.categoryId === addProduct?.category?.id,
+    );
+  }, [addProduct.category]);
+
+  return (
+    <section className={'p-[16px]'}>
+      <h1 className={'text-[24px] font-bold text-text_com mb-4'}>
+        Подати оголошення
+      </h1>
+      <div id={'product-name'} className={'w-2/3 mb-6'}>
+        <Label label={'Назва товару'} />
+        <input
+          placeholder={
+            "Введіть ім'я використовуючи українську або латинську абетку"
+          }
+          className={inputStyle}
+          id={'name'}
+          required
+          value={addProduct.name}
+          onChange={e => {
+            dispatch(SET_FIELD({ ['name']: e.target.value }));
+          }}
+        />
+      </div>
+
+      <div className={'w-full flex gap-6 mb-6'} id={'row-select-category'}>
+        <div id={'category'} className={'w-1/3'}>
+          <Label label={'Вибір Категорії'} />
+          <Select
+            value={addProduct.category}
+            onChange={val => {
+              console.log(val);
+              dispatch(
+                SET_FIELD({
+                  ['category']: {
+                    value: val?.value ?? '',
+                    label: val?.label ?? '',
+                    id: val?.id ?? '',
+                  },
+                }),
+              );
+            }}
```

Comment:
Move it to a separate method. It applies to all similar methods in onChange in this component.

Reply:
done.